mender/templates/_helpers.tpl

```diff
@@ -53,3 +53,20 @@ Redis address
   {{- end }}
 {{- end }}
 
+{{/*
+MongoDB URI
+*/}}
+{{- define "mongodb_uri" }}
+  {{- if and ( not .Values.mongodb.enabled ) ( .Values.global.mongodb.URL ) }}
+    {{- printf .Values.global.mongodb.URL | b64enc }}
+  {{- else if and ( .Values.mongodb.enabled ) ( not .Values.mongodb.auth.enabled ) }}
+    {{- if eq .Values.mongodb.architecture "replicaset" }}
+    {{- printf "mongodb://%s-%s" .Release.Name "mongodb-headless" | b64enc }}
+    {{- else }}
+      {{- printf "mongodb://%s-%s" .Release.Name "mongodb" | b64enc }}
+    {{- end }}
+  {{- else }}
```

Comment:
This breaks in any cases: we're using a secret named `mongodb-common` with the data `MONGO_URL`: this is not compatible with the bitnami way: 
they let you define a service name (overridable) to point to.

The `mongodb_uri` template is just for creating the value for `MONGO_URL`, not defining the k8s service name.
Anyway I could add some guardrails to prevent misconfigurations

Reply:
Added the `.Subchart.[chartName]`  feature: https://github.com/helm/helm/pull/9957
It's available from Helm 3.7, though, but it's really powerful.